### DISCUSSION TITLE
Backup and restore plans

### DISCUSSION THREAD

**Discussion by scipe** [milvus-io/milvus#15621] [Q&A and General discussion]
Hi folks. I just wanna have the document that will describe backups and restores steps if something goes wrong.

But I can not really found anything in documentation regarding to backups and found only one issue on Github and seems like it's was for Milvus 1.x.

So do we have any documentation about backups?

I'd love to see something like:
Backup:
- We have to backup etcd volumes
- We have to backup or create revisions for s3/minio
Restore:
- Stop all services
- Restore the right version of etcd volumes from snapshots/backups
- Restore the right version of s3/minio files
- Start all services

**Comment by xiaofan-luan** [comment]
umm, we don't support backup support yet. you may do it your self but it's a little bit difficult because you also need to consider data in stream storage(pulsar). we may want to implement one I guess

**Reply by scipe** [reply]
Oh I see. Is any tickets or plans on roadmaps regarding to backups?

**Reply by xiaofan-luan** [reply]
The team is working on bulkload data into the system, this should be done in 2.1(April)
On 2.2(June) we may want to support export data out of the system,  then we can implement the functionality of backup by simply exporting and importing the data.

**Reply by xiaofan-luan** [reply]
we have a roadmap and it's a little bit outdated, I will update it in a few days.
If anyone want to contribute on those features that would be great
